### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by williamprofit**

### CODE REVIEW

File: src/app/parser/parser.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/* eslint-disable max-classes-per-file */
```
Comment (lancelotblanchard) [inline src/app/parser/parser.ts]: Can we maybe disable this error in `.eslintrc.js`? It seems to be used in several files.

File: src/app/parser/parser.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    public readonly args: string[],
+  ) {
+  }
+}
+
+/**
+ * Given a Python file, returns the indentation character sequence
+ * @param file file to get the indentation for
+ */
+const getIndentation = (file: string): string | null => {
```
Comment (lancelotblanchard) [inline src/app/parser/parser.ts]: Here and below, is there a reason why we use `const funcName = () => {}` instead of `function funcName() {}`?

**Review by lancelotblanchard — DISMISSED** [review]
Logic and tests look really good, congrats! 👏 I just left two minor comments, but once these are addressed, it looks good to merge!

File: src/app/parser/parser.ts (hunk 2 above)
Comment (williamprofit) [reply src/app/parser/parser.ts]: this is ES6 format which is more recent, can use the old one

Comment (lancelotblanchard) [reply src/app/parser/parser.ts]: Oh ok, let's leave it that way then!

**Review by lancelotblanchard — APPROVED** [review]
😮

**Pull request merged**